docs/content/tutorial/building-an-asset-graph.mdx

```diff
@@ -0,0 +1,235 @@
+    # Since we're saving the asset, this converts the word cloud into an image
+    buffer = BytesIO()
+    plt.savefig(buffer, format="png")
+    image_data = base64.b64encode(buffer.getvalue())
+
+    return image_data # returns the image instead of showing it with `plt.show()`
+```
+
+---
+
+## Educating users with metadata
+
+Software-defined assets can be enriched with different types of metadata. Anything can be used as metadata for an SDA. Common details to add are:
+
+- Statistics about the data, such as row counts or other data profiling
+- Test results or assertions about the data
+- Images or tabular previews of the asset
+- Information about who owns the asset, where it's stored, and links to external documentation
+
+The following code adds a row count and a preview of the `topstories` asset. It uses the <PyObject module="dagster" object="Output" displayText="Output"/> class from Dagster to bundle the asset and its metadata together. Update your code for the `topstories` asset to match the changes below.
+
+```python
+from dagster import asset, get_dagster_logger, Output, MetadataValue
+# add an import to the Output and MetadataValue classes from the dagster library
+
+@asset(group_name="hackernews")
+def topstories(context, topstory_ids):
+    results = []
+    for item_id in topstory_ids:
+        item = requests.get(f"<https://hacker-news.firebaseio.com/v0/item/{item_id}.json>").json()
+        results.append(item)
+
+        if len(results) % 20 == 0:
+            context.log.info(f"Got {len(results)} items so far.")
+
+    df = pd.DataFrame(results)
+
+    return Output(  # The return value is updated to wrap it in `Output` class
+        value=df,   # The original df is passed in with the `value` parameter
+        metadata={
+            "num_records": len(df), # Metadata can be any key-value pair
+            "preview": MetadataValue.md(df.head().to_markdown()),
+            # The `MetadataValue` class has useful static methods to build Metadata
+        }
+    )
+```
+
+Reload the definitions and re-materialize your assets. The metadata can then be seen in the following places:
+
+- In the **Asset graph** page, click on an asset and its metadata will be shown in the right sidebar:
+
+<Image
+alt="the asset's materialization metadata"
+src="/images/tutorial/asset-metadata.png"
+width={1152}
+height={666}
+/>
+
+- In the **Asset Catalog's** page for the `topstories` asset:
+
+<Image
+alt="the asset's materialization metadata"
+src="/images/tutorial/asset-metadata.png"
+width={1152}
+height={667}
+/>
+
+### Metadata and Markdown
+
+The data frame was embedded into the asset's metadata with Markdown. Any valid Markdown snippet can be stored and rendered in the Dagster UI, including images. By embedding the word cloud image as metadata, the word cloud asset is also visible to your team from within the UI.
+
+Below is code that changes shows how to add an image in asset metadata. Add the following to the end of your `assets.py`:
```

Comment:
💭 I didn't catch this when we were editing in Notion, but not showing the wordcloud here feels like a miss. Thoughts on adding it, if 1. We didn't already decide against it and I missed it, and 2. It won't take too much time?

Reply:
and added!